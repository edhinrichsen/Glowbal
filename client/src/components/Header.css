header {
  display: flex;
  /* font-weight: bold; */
}

header ul {
  margin-top: auto !important;
  margin-bottom: auto !important;
  border: 0 !important;
}

a.navTitle {
  color: rgba(0, 0, 0, 0.85);
}

.ant-input-group > .ant-input:first-child,
.ant-input-group-addon:first-child {
  height: 40px;
  border-radius: 100px 0px 0px 100px !important;
  border: var(--border);
  box-shadow: inset 0px 3px 5px rgba(0, 0, 0, 0.1), var(--drop-shadow);
  background-color: rgba(0, 0, 0, 0.03);
  /* box-shadow: var(--drop-shadow); */
}

button.ant-btn.ant-btn-primary.ant-input-search-button {
  position: relative;
  z-index: 1;
  border-radius: 0px 100px 100px 0px !important;
}

.ant-input-group-addon {
  border-top-right-radius: 100px;
  border-bottom-right-radius: 100px;
}

.ant-menu {
  border-color: var(--color-primary) !important;
  color: var(--color-primary) !important;
  background-color: var(--background-color-app);
}
.ant-menu-horizontal > .ant-menu-item-selected a {
  background: var(--color-app);
  background: var(--gradient);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  text-shadow: var(--drop-shadow);
}
