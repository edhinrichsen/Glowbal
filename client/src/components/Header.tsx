import React, { useState } from 'react';
import { NavLink, BrowserRouter, Link } from 'react-router-dom';

import { PageHeader, Avatar, Menu, Input, Space } from 'antd';
import { UserOutlined } from '@ant-design/icons';

import './Header.css';

const { Search } = Input;

// From antd spec
type MenuClickEvent = {
    item: string,
    key: string,
    keyPath: string, 
    domEvent: Event
}

type HeaderProps = {
    pageKey: string,
}

export function Header({ pageKey }: HeaderProps) {

    const [ currentPageSelection, setCurrentPageSelection ] = useState(pageKey);

    const handleMenuClick = (event: MenuClickEvent) => {
        setCurrentPageSelection(event.key);
    }

    return(
        <div>
            <header>
                    <PageHeader
                      className="site-page-header"
                      title={<Link to="/" className="navTitle"> <span role="img" aria-hidden="true">📝</span> ePortfolio </Link>}
                      extra={
                        <div>
                            <Space>
                                <Search style={{ width: 300 }}enterButton />
                                <Avatar icon={<UserOutlined/>} />
                            </Space>
                        </div>
                      } 
                    />

                <Menu
                  onClick={() => handleMenuClick}
                  selectedKeys={[currentPageSelection]}
                  mode="horizontal"
                > 

                      <Menu.Item key="home" icon={<span role="img" aria-hidden="true">🏠 </span>}>
                        <NavLink exact to="/">Home</NavLink>
                      </Menu.Item>

                      <Menu.Item key="profile" icon={<span role="img" aria-hidden="true">👨‍💼 </span>}>
                        <NavLink exact to="/profile">Profile</NavLink> {/* TODO: Protected route once authentication is set up */}
                      </Menu.Item>

                </Menu>

            </header>
        </div>
    );
}

export default Header;