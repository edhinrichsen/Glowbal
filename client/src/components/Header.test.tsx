import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';

describe('Testing Header', () => {

  it('renders home without crashing', () => {
    const component = shallow(<Header pageKey="home"/>);
    expect(component).toMatchSnapshot();
    component.unmount();
  });
    
  it('renders profile without crashing', () => {
    const component = shallow(<Header pageKey="profile"/>);
    expect(component).toMatchSnapshot();
    component.unmount();
  });
    
})
