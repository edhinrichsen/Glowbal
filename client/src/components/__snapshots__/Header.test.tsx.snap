// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Header renders home without crashing 1`] = `
<div>
  <header>
    <PageHeader
      className="site-page-header"
      title={
        <ForwardRef
          className="navTitle"
          to="/"
        >
           
          <div
            className="gradient"
            style={
              Object {
                "fontSize": "1.2em",
                "marginLeft": "40px",
              }
            }
          >
            <strong>
              Glowbal
            </strong>
          </div>
        </ForwardRef>
      }
    />
    <Menu
      mode="horizontal"
      onClick={[Function]}
      selectedKeys={
        Array [
          "home",
        ]
      }
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <MenuItem
        icon={
          <span
            aria-hidden="true"
            role="img"
          >
            <ForwardRef(HomeOutlined) />
          </span>
        }
        key="home"
      >
        <NavLink
          exact={true}
          to="/"
        >
          Home
        </NavLink>
      </MenuItem>
      <MenuItem
        icon={
          <span
            aria-hidden="true"
            role="img"
          >
            <ForwardRef(UserOutlined) />
          </span>
        }
        key="profile"
      >
        <NavLink
          exact={true}
          to="/profile"
        >
          Profile
        </NavLink>
         
      </MenuItem>
    </Menu>
    <Space
      style={
        Object {
          "marginLeft": "auto",
          "marginRight": "20px",
        }
      }
    >
      <Tooltip
        arrowPointAtCenter={false}
        autoAdjustOverflow={true}
        mouseEnterDelay={0.1}
        mouseLeaveDelay={0.1}
        placement="top"
        title="Toggle Dark Mode"
        transitionName="zoom-big-fast"
      >
        <div
          className="toggle-dark-mode gradient"
          onClick={[Function]}
        />
      </Tooltip>
        
      <Search
        enterButton={true}
        onSearch={[Function]}
        style={
          Object {
            "width": 300,
          }
        }
      />
      <LoginButton />
    </Space>
  </header>
</div>
`;

exports[`Testing Header renders profile without crashing 1`] = `
<div>
  <header>
    <PageHeader
      className="site-page-header"
      title={
        <ForwardRef
          className="navTitle"
          to="/"
        >
           
          <div
            className="gradient"
            style={
              Object {
                "fontSize": "1.2em",
                "marginLeft": "40px",
              }
            }
          >
            <strong>
              Glowbal
            </strong>
          </div>
        </ForwardRef>
      }
    />
    <Menu
      mode="horizontal"
      onClick={[Function]}
      selectedKeys={
        Array [
          "profile",
        ]
      }
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <MenuItem
        icon={
          <span
            aria-hidden="true"
            role="img"
          >
            <ForwardRef(HomeOutlined) />
          </span>
        }
        key="home"
      >
        <NavLink
          exact={true}
          to="/"
        >
          Home
        </NavLink>
      </MenuItem>
      <MenuItem
        icon={
          <span
            aria-hidden="true"
            role="img"
          >
            <ForwardRef(UserOutlined) />
          </span>
        }
        key="profile"
      >
        <NavLink
          exact={true}
          to="/profile"
        >
          Profile
        </NavLink>
         
      </MenuItem>
    </Menu>
    <Space
      style={
        Object {
          "marginLeft": "auto",
          "marginRight": "20px",
        }
      }
    >
      <Tooltip
        arrowPointAtCenter={false}
        autoAdjustOverflow={true}
        mouseEnterDelay={0.1}
        mouseLeaveDelay={0.1}
        placement="top"
        title="Toggle Dark Mode"
        transitionName="zoom-big-fast"
      >
        <div
          className="toggle-dark-mode gradient"
          onClick={[Function]}
        />
      </Tooltip>
        
      <Search
        enterButton={true}
        onSearch={[Function]}
        style={
          Object {
            "width": 300,
          }
        }
      />
      <LoginButton />
    </Space>
  </header>
</div>
`;
